docs/src/content/docs/reference/plugins.md

```diff
@@ -41,12 +47,100 @@ A plugin must provide a unique name that describes it. The name is used when [lo
 
 ## `hooks`
 
-Hooks are functions which Starlight calls to run plugin code at specific times. Currently, Starlight supports a single `setup` hook.
+Hooks are functions which Starlight calls to run plugin code at specific times.
+
+To get the type of a hook's arguments, use the `HookParameters` utility type and pass in the hook name.
+In the following example, the `configSetup` function `options` parameter is typed to match the parameters of the `config:setup` hook:
 
-### `hooks.setup`
+```ts
+import type { HookParameters } from '@astrojs/starlight/types';
 
-Plugin setup function called when Starlight is initialized (during the [`astro:config:setup`](https://docs.astro.build/en/reference/integrations-reference/#astroconfigsetup) integration hook).
-The `setup` hook can be used to update the Starlight configuration or add Astro integrations.
+function configSetup(options: HookParameters['config:setup']) {
+  options.useTranslations('en');
+}
+```
+
+### `i18n:setup`
+
+Plugin internationalization setup function called when Starlight is initialized (during the [`astro:config:setup`](https://docs.astro.build/en/reference/integrations-reference/#astroconfigsetup) integration hook).
+The `i18n:setup` hook can be used to inject translations strings for the plugin in various locales.
+These translations will be available in the [`config:setup`](#configsetup) hook and plugin UI.
```

Comment:
Suggesting a small tweak here. “plugin UI” sounds a bit like some kind of UI for managing a plugin somehow to me, so wanted to try to emphasise “components” to help with understanding that this is in your website’s UI and how to access them.

```suggestion
The `i18n:setup` hook can be used to inject translation strings so a plugin can support different locales.
These translations will be available via [`useTranslations()`](#usetranslations) in the `config:setup` hook and in UI components via [`Astro.locals.t()`](/guides/i18n/#using-ui-translations).
```